File: JavaImpl/src/main/java/org/example/BigONotation/BubbleSortExample.java
package org.example.BigONotation;

import java.util.Arrays;

public class BubbleSortExample {
    public static void main(String[] args) {
        int[] array = {5, 3, 8, 1, 2};
        bubbleSort(array);
        System.out.println("Sorted: " + Arrays.toString(array)); // O(n²)
    }

    public static void bubbleSort(int[] array) {
        int n = array.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    // Swap
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                }
            }
        }
    }
}

File: KotlinImpl/src/main/kotlin/BigONotation/BubbleSortExample.kt
package org.example.BigONotation

fun main() {
    val array = intArrayOf(5, 3, 8, 1, 2)
    bubbleSort(array)
    println("Sorted: ${array.joinToString(", ")}") // O(n²)
}

fun bubbleSort(array: IntArray) {
    val n = array.size
    for (i in 0 until n - 1) {
        for (j in 0 until n - i - 1) {
            if (array[j] > array[j + 1]) {
                // Swap
                val temp = array[j]
                array[j] = array[j + 1]
                array[j + 1] = temp
            }
        }
    }
}
